src/github.com/matrix-org/dendrite/roomserver/input/consumer.go

```diff
@@ -0,0 +1,91 @@
+// Package input contains the code that writes
+package input
+
+import (
+	"github.com/matrix-org/dendrite/roomserver/types"
+	sarama "gopkg.in/Shopify/sarama.v1"
+)
+
+// A ConsumerDatabase has the storage APIs needed by the consumer.
+type ConsumerDatabase interface {
+	// PartitionOffsets returns the offsets the consumer has reached for each partition.
+	PartitionOffsets(topic string) ([]types.PartitionOffset, error)
+	// SetPartitionOffset records where the consumer has reached for a partition.
+	SetPartitionOffset(topic string, partition int32, offset int64) error
+}
+
+// An ErrorHandler handles the errors encountered by the consumer.
+type ErrorHandler interface {
+	OnError(err error)
```

Comment:
What's the intent behind this? Why aren't we just logging errors or whatever? There's no return type, so it's not like the entity which listens for this can *do* anything about it.